test/FunctionalTests/Microsoft.OData.Core.Tests/UriParser/Parsers/CustomUriLiteralPrefixesTests.cs

```diff
@@ -327,10 +327,30 @@ public void CustomUriLiteralPrefix_CannotParseWithCustomLiteralPrefix_IfBuiltInP
         [Fact]
         public void CustomUriLiteralPrefix_CannotParseTypeWithWrongLiteralPrefix()
         {
+            // Ensure the prefix under test is registered. Handle the exception gracefully if
+            // the prefix is already registered.
+            IEdmTypeReference booleanTypeReference = EdmCoreModel.Instance.GetBoolean(false);
+
             try
             {
-                IEdmTypeReference booleanTypeReference = EdmCoreModel.Instance.GetBoolean(false);
-                CustomUriLiteralPrefixes.AddCustomLiteralPrefix(CustomUriLiteralParserUnitTests.BOOLEAN_LITERAL_PREFIX, booleanTypeReference);
+                CustomUriLiteralPrefixes.AddCustomLiteralPrefix(
+                    CustomUriLiteralParserUnitTests.BOOLEAN_LITERAL_PREFIX, booleanTypeReference);
+            }
+            catch (ODataException e)
+            {
+                if (!String.Equals(e.Message, Strings.CustomUriTypePrefixLiterals_AddCustomUriTypePrefixLiteralAlreadyExists(
+                    CustomUriLiteralParserUnitTests.BOOLEAN_LITERAL_PREFIX)))
+                {
+                    // unexpected exception, re-throw.
+                    throw;
+                }
+
+                // Swallow the exception since it is due to trying to register a prefix that is already added.
+            }
+
+            try
+            {
+

```

Comment:
nit: extra empty line

Reply:
removed.

---
In reply to: [141484785](https://github.com/OData/odata.net/pull/962#discussion_r141484785) [](ancestors = 141484785)